lib/streamlit/cli.py

```diff
@@ -117,12 +117,22 @@ def main_hello():
 @main.command("run")
 @click.argument("file_or_url", required=True)
 @click.argument("args", nargs=-1)
-def main_run(file_or_url, args):
+def main_run(file_or_url, args=None):
     """Run a Python script, piping stderr to Streamlit.
     The script can be local or it can be an url. In the
     latter case, streamlit will download the script to a
     temporary file and runs this file.
     """
+    import subprocess
+
+    cmd_line_as_list = [
+        click.get_current_context().parent.command_path,
+        click.get_current_context().command.name,
+        file_or_url,
+    ]
```

Comment:
I think this make more sense than grabbing from `sys.argv` even if I have to "re-build" it here manually. `click` API is also a bit obscure. These above seem to work and a test is included.

Reply:
What if the use passed in some flags or args? Does this still work?

For example:
```
$ streamlit --server.port=8000 run foo.py arg1 arg2
```

The goal is for the command line above to appear in the dialog as the user typed it.

Reply:
Right, I think I will have to use `click.get_os_args()` to capture that. Will change.